backend/dataall/modules/redshift_datasets/services/redshift_dataset_service.py

```diff
@@ -45,6 +46,46 @@
 
 
 class RedshiftDatasetService:
+    _interfaces: List[DatasetServiceInterface] = []
+
+    @classmethod
+    def register(cls, interface: DatasetServiceInterface):
+        cls._interfaces.append(interface)
```

Comment:
more of a question rather than a comment: why do we need to keep a list? don't we always register just one?

Reply:
So far yes; we communicate from redshift-shares to redshift-dataset. I imagined other consumption modules that would need to interact with the dataset module. For example a data-pipeline that depends on datasets and the dataset should not be deleted if it has pipelines